app/models/singly_linked_list.rb

```diff
@@ -0,0 +1,77 @@
+class SinglyLinkedList
+  include Enumerable
```

Comment:
The code that does this is written in C (I think it's [this function](https://github.com/ruby/ruby/blob/6ee8ec70696260fb0ac2ab4bf13f40ff003a8995/vm_eval.c#L1147)), but you can get the same behavior with Ruby code that looks like:

```ruby
def first(max)
  count = 0
  acc = []

  each do |n|
    break if count == max
    acc << n
    count += 1
  end

  acc
end
```